src/core/common/tlvs.cpp

```diff
@@ -273,13 +256,12 @@ template Error Tlv::FindUintTlv<uint32_t>(const Message &aMessage, uint8_t aType
 
 Error Tlv::FindTlv(const Message &aMessage, uint8_t aType, void *aValue, uint16_t aLength)
 {
-    Error    error;
-    uint16_t offset;
-    uint16_t length;
+    Error       error = kErrorNotFound;
+    OffsetRange offsetRange;
 
-    SuccessOrExit(error = FindTlvValueOffset(aMessage, aType, offset, length));
-    VerifyOrExit(length >= aLength, error = kErrorParse);
-    aMessage.ReadBytes(offset, aValue, aLength);
+    SuccessOrExit(FindTlvValueOffsetRange(aMessage, aType, offsetRange));
+    SuccessOrExit(aMessage.Read(offsetRange, aValue, aLength));
+    error = kErrorNone;
```

Comment:
```suggestion
    SuccessOrExit(error = FindTlvValueOffsetRange(aMessage, aType, offsetRange));
    SuccessOrExit(error = aMessage.Read(offsetRange, aValue, aLength));
```
What about writing like this?

Reply:
Make sense. Thanks!